integrations/terraform/hashicorp-vault/README.md

```diff
@@ -0,0 +1,219 @@
+
+```hcl
+provider "vault" {
+  address = "http://localhost:8200"
+  token = "<Root-Token>"
+}
+```
+
+Replace `<Root-Token>` with the actual Root Token of your Vault server.
+
+After that initialize the Materialize provider:
+
+```hcl
+provider "materialize" {
+  host     = local.materialize_host
+  username = local.materialize_username
+  password = local.materialize_password
+  port     = 6875
+  database = "materialize"
+}
+```
+
+You'll need to set the `local` variables for `materialize_host`, `materialize_username`, and `materialize_password`:
+
+```hcl
+locals {
+  # The Materialize Cloud details
+  materialize_host     = "cloud.materialize.com"
+  materialize_username = "<Materialize-Username>"
+  materialize_password = "<Materialize-Password>"
+  # The Vault root token
+  vault_token          = "<Root-Token>"
+}
+```
+
+Replace `<Materialize-Username>` and `<Materialize-Password>` with the actual username and password of your Materialize Cloud account.
+
+## Retrieving a Secret from Vault
+
+Now you can use the `vault_generic_secret` data source to retrieve the secret from Vault:
+
+```hcl
+data "vault_generic_secret" "materialize_password" {
+  path = "secret/materialize"
+}
+```
+
+Finally, you can use the secret retrieved from Vault in your `materialize_secret` resource:
+
+```hcl
+resource "materialize_secret" "example_secret" {
+  name  = "pgpass"
+  value = data.vault_generic_secret.materialize_password.data["pgpass"]
+}
+```
+
+## Running the Terraform Script
+
+Now that you have the Terraform script ready, you can run the following commands to initialize and apply the script:
+
+```bash
+# Initialize the Terraform configuration:
+terraform init
+
+# Apply the Terraform configuration:
+terraform apply
+```
+
+Output:
+
+```bash
+Plan: 1 to add, 0 to change, 0 to destroy.
+
+Do you want to perform these actions?
+  Terraform will perform the actions described above.
+  Only 'yes' will be accepted to approve.
+
+  Enter a value: yes
+
+materialize_secret.example_secret: Creating...
+materialize_secret.example_secret: Creation complete after 3s [id=u111]
+
+Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
+```
+
+Once the script has been applied, you can verify that the secret has been created in Materialize Cloud:
+
+```sql
+SHOW SECRETS;
+```
+
+## Using the Secret in Connections
+
+Once the secret has been created, you can use it in your `CREATE CONNECTION` statements. In this example, we'll create a connection to a Postgres instance using the secret we created using the `materialize_connection_postgres` resource:
+
+```hcl
+# Create a Postgres Connection
+resource "materialize_connection_postgres" "example_postgres_connection" {
+  name = "example_postgres_connection"
+  host = "instance.foo000.us-west-1.rds.amazonaws.com"
+  port = 5432
+  user {
+    text = "pguser"
+  }
+  password {
+    name          = materialize_secret.example_secret.name
+    database_name = "materialize"
```

Comment:
You should be able to get the `database_name` and `schema_name` with references to the secret as well
```
    database_name = materialize_secret.example_secret.database_name
    schema_name   = materialize_secret.example_secret.schema_name
```